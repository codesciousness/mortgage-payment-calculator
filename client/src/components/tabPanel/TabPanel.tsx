import React from 'react';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Paper from '@mui/material/Paper';

type TabPanelProps = {
    children: React.ReactNode,
    value: number,
    index: number
};

const TabPanel = ({ children, value, index }: TabPanelProps): JSX.Element => {

    return (
        <div
            role="tabpanel"
            hidden={value !== index}
            id={`tabpanel-${index}`}
            aria-labelledby={`tab-${index}`}
        >
            {value === index && (
                <Box component={Paper} elevation={4} sx={{ p: 3 }}>
                    <Typography>{children}</Typography>
                </Box>
            )}
        </div>
    );
};

export default TabPanel;