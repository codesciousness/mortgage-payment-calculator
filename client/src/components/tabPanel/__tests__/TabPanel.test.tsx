import React from 'react';
import { render, screen } from '@testing-library/react';
import { create } from 'react-test-renderer';
import TabPanel from '../TabPanel';

const children = <section data-testid="TabPanel-element">Test Node</section>;
const valueVisible = 0;
const valueHidden = 1;
const index = 0;

describe('<TabPanel />', () => {
    test('TabPanel component is defined', () => {
        expect(<TabPanel children={children} value={valueHidden} index={index}/>).toBeDefined();
    });
    test('component matches the snapshot', () => {
        const component = create(<TabPanel children={children} value={valueVisible} index={index}/>);
        expect(component.toJSON()).toMatchSnapshot();
    });
    test('displays children when the value equals the index', () => {
        render(<TabPanel children={children} value={valueVisible} index={index}/>);
        expect(screen.getByTestId('TabPanel-element')).toBeInTheDocument();
        expect(screen.getByText('Test Node')).toBeInTheDocument();
    });
    test('does not display children when the value is not equal to the index', () => {
        render(<TabPanel children={children} value={valueHidden} index={index}/>);
        expect(screen.queryByTestId('TabPanel-element')).not.toBeInTheDocument();
        expect(screen.queryByText('Test Node')).not.toBeInTheDocument();
    });
});