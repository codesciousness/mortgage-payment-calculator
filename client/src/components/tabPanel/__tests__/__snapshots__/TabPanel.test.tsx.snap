// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabPanel /> component matches the snapshot 1`] = `
<div
  aria-labelledby="tab-0"
  hidden={false}
  id="tabpanel-0"
  role="tabpanel"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation4 MuiBox-root css-1ggs94k-MuiPaper-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      <section
        data-testid="TabPanel-element"
      >
        Test Node
      </section>
    </p>
  </div>
</div>
`;
